### PR TITLE
Move common master upgrade playbooks to openshift-master component

### PR DESCRIPTION
**Pull request by mtnbikenc**
* Organize master upgrade playbooks for consistency
* Additionally removes unused symlinks

### CODE REVIEW

File: playbooks/openshift-master/private/upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+# The main master upgrade play. Should handle all changes to the system in one pass, with
+# support for optional hooks to be defined.
+- name: Upgrade master
+  hosts: oo_masters_to_config
+  serial: 1
+  roles:
+  - openshift_facts
+  tasks:
+  # Run the pre-upgrade hook if defined:
+  - debug: msg="Running master pre-upgrade hook {{ openshift_master_upgrade_pre_hook }}"
```
Comment (michaelgugino) [inline playbooks/openshift-master/private/upgrade.yml]: Do users provide an absolute path here?

File: playbooks/openshift-master/private/upgrade.yml [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+
+  - include_tasks: "{{ openshift_master_upgrade_pre_hook }}"
+    when: openshift_master_upgrade_pre_hook is defined
+
+  - import_role:
+      name: openshift_master
+      tasks_from: upgrade.yml
+
+  - name: Setup and enable bootstrapping options
+    include_tasks: ../../../openshift-master/private/tasks/enable_bootstrap.yml
```
Comment (michaelgugino) [inline playbooks/openshift-master/private/upgrade.yml]: Let's cleanup these include_tasks if they are in this directory.

File: playbooks/openshift-master/private/upgrade.yml [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+      reconcile_completed: "{{ hostvars
+                                 | lib_utils_oo_select_keys(groups.oo_masters_to_config)
+                                 | lib_utils_oo_collect('inventory_hostname', {'reconcile_complete': true}) }}"
+  - set_fact:
+      reconcile_failed: "{{ groups.oo_masters_to_config | difference(reconcile_completed) | list }}"
+  - fail:
+      msg: "Upgrade cannot continue. The following masters did not finish reconciling: {{ reconcile_failed | join(',') }}"
+    when: reconcile_failed | length > 0
+
+- name: Upgrade Docker on dedicated containerized etcd hosts
```
Comment (michaelgugino) [inline playbooks/openshift-master/private/upgrade.yml]: This is not part of masters.

File: playbooks/openshift-master/private/upgrade.yml (hunk 1 above)
Comment (mtnbikenc) [reply playbooks/openshift-master/private/upgrade.yml]: Yes.  See HOOKS.md.

File: playbooks/openshift-master/private/upgrade.yml (hunk 3 above)
Comment (mtnbikenc) [reply playbooks/openshift-master/private/upgrade.yml]: Do you think this should be moved to the etcd upgrade playbooks?

Comment (michaelgugino) [reply playbooks/openshift-master/private/upgrade.yml]: yeah and probably should not be included in master's upgrade steps at all.  Should be an entirely separate concern.

Specifying '!oo_nodes_to_upgrade' will ensure that any etcd hosts that are also master hosts will not be run here anyway.

**Comment by mtnbikenc** [comment]
Rebased and updated.

**Pull request merged**